sdks/java/core/src/main/java/org/apache/beam/sdk/util/RowJson.java

```diff
@@ -375,6 +400,9 @@ private void writeRow(Row row, Schema schema, JsonGenerator gen) throws IOExcept
       for (int i = 0; i < schema.getFieldCount(); ++i) {
         Field field = schema.getField(i);
         Object value = row.getValue(i);
+        if (ignoreNullsOnWrite && value == null){
+          continue;
+        }
```

Comment:
This should also check `field.getType().getNullable()` like the other conditional. If we get a null for a non-nullable field we should fail loudly rather than silently dropping it.